slackbot.js

```diff
@@ -9,49 +9,52 @@ function withDefaultOptions (env, options) {
 
 function Slackbot (transport, env, options) {
   var mergedOptions = withDefaultOptions(env, options);
-  var rtm = new transport.RtmClient(env.vars.SLACK_TOKEN, mergedOptions);
-  var web = new transport.WebClient(env.vars.SLACK_TOKEN, mergedOptions);
-  var logger = helpers.logger('Slackbot', mergedOptions.logLevel === 'verbose');
+  this.rtm = new transport.RtmClient(env.vars.SLACK_TOKEN, mergedOptions);
+  this.web = new transport.WebClient(env.vars.SLACK_TOKEN, mergedOptions);
+  this.logger = helpers.logger('Slackbot', mergedOptions.logLevel === 'verbose');
+}
 
-  function includeChannel (next) {
-    var yak = logger.sub('includeChannel');
-    return function addChannelToResponse (response) {
+Slackbot.prototype.includeChannel = function (next) {
+  var yak = this.logger.sub('includeChannel');
+  return (function (web) {
+    return function (response) {
       if (!response.channel_id) {
         yak('no channel_id found, will not place call', response);
         return next(response);
       }
       yak('channel_id found in response, get info for', response.channel_id);
       return web.channels.info(
         response.channel_id,
-        helpers.handleError(function (channelRes) {
-          yak('channel response for', response.channel_id, channelRes);
-          next(response, channelRes);
+        helpers.handleError(function (channelResponse) {
+          yak('channel response for', response.channel_id, channelResponse);
+          next(response, channelResponse);
         }));
     };
-  }
-  function getFileInfo (id, next) {
-    var yak = logger.sub('getFileInfo');
-    yak('getting file', id);
-    web.files.info(id, helpers.handleError(function (res) {
-      yak('received file', res.file);
-      next(res.file);
-    }));
-  }
-  function onPinAdded (next) {
-    rtm.on(slack.RTM_EVENTS.PIN_ADDED, includeChannel(function (res, chan) {
-      logger('onPinAdded', res);
-      next.call(this, res, chan);
-    }));
-  }
-  function connect (next) {
-    rtm.on(slack.CLIENT_EVENTS.RTM.AUTHENTICATED, next);
-    rtm.start();
-  }
-  return {
-    getFileInfo: getFileInfo,
-    onPinAdded: onPinAdded,
-    connect: connect
-  };
-}
+  })(this.web);
+};
+
+Slackbot.prototype.getFileInfo = function (id, next) {
+  var yak = this.logger.sub('getFileInfo');
+  yak('getting file', id);
+  this.web.files.info(id, helpers.handleError(function (res) {
+    yak('received file', res.file);
+    next(res.file);
+  }));
+};
+
+Slackbot.prototype.onPinAdded = function (next) {
+  // TODO verify use of 'this' in next.call
+  // TODO verify change in logger logic
```

Comment:
functions that aren't explicitly bound via `.bind`, `.call`, `.apply`, or dot lookup, have `this` set to a common default value. in strict mode it's `null`. this project doesn't have `'use strict';` at the top of every file, so it's in "loose mode", and the default value is the "global object", which in in Node is just an object with every global variable, e.g. `process`, hanging off it. in the browser it's `Window`.